airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java

```diff
@@ -142,7 +160,11 @@ public static List<AutoCloseableIterator<AirbyteMessage>> getCdcReadIterators(fi
    * Determines the streams to sync for initial primary key load. These include streams that are (i) currently in primary key load (ii) newly added
    * incremental streams.
    */
-  public static InitialLoadStreams streamsForInitialPrimaryKeyLoad(final CdcStateManager stateManager, final ConfiguredAirbyteCatalog fullCatalog) {
+  public static InitialLoadStreams streamsForInitialPrimaryKeyLoad(final CdcStateManager stateManager, final ConfiguredAirbyteCatalog fullCatalog, final boolean savedOffsetStillPresentOnServer) {
+    if (!savedOffsetStillPresentOnServer) {
+      return new InitialLoadStreams(fullCatalog.getStreams(), new HashMap<>());
```

Comment:
Let's only include streams that are in INCREMENTAL mode to be safe here. I do think we also filter downstream, but better to be expliciti IMO